pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -39,34 +40,72 @@ type framework struct {
 var _ = Framework(&framework{})
 
 // NewFramework initializes plugins given the configuration and the registry.
-func NewFramework(r Registry, _ *runtime.Unknown) (Framework, error) {
+func NewFramework(r Registry, plugins config.Plugins, args []config.PluginConfig) (Framework, error) {
 	f := &framework{
 		registry:         r,
 		nodeInfoSnapshot: cache.NewNodeInfoSnapshot(),
 		plugins:          make(map[string]Plugin),
 	}
+	// get needed plugins from config
+	pg := pluginsNeeded(plugins)
+	if len(pg) == 0 {
+		return f, nil
+	}
 
-	// TODO: The framework needs to read the scheduler config and initialize only
-	// needed plugins. In this initial version of the code, we initialize all.
+	var pc *runtime.Unknown
+	pluginConfig := pluginNameToConfig(args)
 	for name, factory := range r {
-		// TODO: 'nil' should be replaced by plugin config.
-		p, err := factory(nil, f)
+		// initialize only needed plugins
+		if _, ok := pg[name]; !ok {
+			continue
+		}
+
+		// find the config args of a plugin
+		if v, ok := pluginConfig[name]; ok {
+			pc = v
+		} else {
+			pc = nil
```

Comment:
This if-else is redundant because `pluginConfig[name]` evaluates to `nil` for any key that isn't in the map.

Reply:
Updated. PTAL.
Thanks!!